src/app/modules/myinfo/myinfo.util.ts

```diff
@@ -359,6 +360,37 @@ export const extractMyInfoCookie = (
   return err(new MyInfoMissingAccessTokenError())
 }
 
+/**
+ * Checks if myInfoCookie is successful and returns the result.
+ * This function acts as a discriminator so that the type of the cookie is encoded in its type
+ * @param cookie the cookie to
+ * @returns ok(cookie) the successful myInfoCookie
+ * @returns err(cookie) the errored cookie
+ */
+export const extractSuccessfulCookie = (
+  cookie: MyInfoCookiePayload,
+): Result<MyInfoSuccessfulCookiePayload, MyInfoCookieStateError> =>
+  cookie.state === MyInfoCookieState.Success
+    ? ok(cookie)
+    : err(new MyInfoCookieStateError())
+
+/**
+ * Extracts a successful myInfoCookie from a request's cookies
+ * @param cookies Cookies in a request
+ * @return ok(cookie) the successful myInfoCookie
+ * @return err(MyInfoMissingAccessTokenError) if myInfoCookie is not present on the request
+ * @return err(MyInfoCookieStateError) if the extracted myInfoCookie was in an error state
+ */
+export const extractSuccessfulMyInfoCookie = (
+  cookies: Record<string, unknown>,
+): Result<
+  MyInfoSuccessfulCookiePayload,
+  MyInfoCookieStateError | MyInfoMissingAccessTokenError
+> =>
+  extractMyInfoCookie(cookies).andThen((cookiePayload) =>
+    extractSuccessfulCookie(cookiePayload),
```

Comment:
I think I would rather inline the logic in `extractSuccessfulCookie`, it's quite confusing to have three different `extractCookie` functions

Reply:
can't - `myInfoMiddleware` depends on `extractCookie` ;--; 

if we extract this out, it will log as an error and clear cookie even when the user doesn't have a cookie set (and this isn't an error)